django/db/backends/postgresql/schema.py

```diff
@@ -29,10 +29,10 @@ def _model_indexes_sql(self, model):
                 # a second index that specifies their operator class, which is
                 # needed when performing correct LIKE queries outside the
                 # C locale. See #12234.
-                if db_type.startswith('varchar'):
+                if db_type.startswith('varchar') and not db_type.endswith('[]'):
```

Comment:
Comment above also needs to be updated to not confuse reader about new adding, which is not related to ticket in comment